packages/nimble-components/src/rich-text-editor-viewer/specs/README.md

```diff
@@ -0,0 +1,335 @@
+
+### Risks and Challenges
+
+-   Visual design and interaction design for both the components are out of scope for the initial release. We will start with the existing
+    [mockup screens](https://www.figma.com/file/Q5SU1OwrnD08keon3zObRX/SystemLink?type=design&node-id=6280-94045) which is designed for
+    comments feature. At this moment, we cannot afford to cover all the corner case of visual design specs but we managed to refer from
+    other visual designs and try to incorporate with the nimble standards.
+-   Due to the immediate requirement for GM (`Comments Feature`), we are not adding any additional enhancements or feature requirements
+    apart from the mentioned features in the initial release.
+-   The mobile view of the component has not been designed yet, and we are actively collaborating with the design team to create basic mockup screens. We will
+    update this spec accordingly as progress is made.
+
+### Prior Art/Examples
+
+Sample Mockup Screenshots
+
+_`nimble-rich-text-editor`_
+
+![Editor](./spec-images/editor-sample.png)
+
+_`nimble-rich-text-viewer`_
+
+![Viewer](./spec-images/viewer-sample.png)
+
+## Design
+
+_nimble-rich-text-editor_
+
+The `nimble-rich-text-editor` will be divided into two sections namely an `editor` section and a `footer` section.
+
+1. `editor` section is the actual text area to add or update rich text contents.
+2. `footer` section consists of `nimble-button` to control each text formatting functionalities like bold, italic etc,. and a
+   `footer-actions` slot element which is typically used to add action buttons to the right bottom of the component.
+
+Here is the basic high level template used in this component:
+
+```html
+<template>
+    <section>
+        <div id="editor"></div>
+    </section>
+    <footer>
+        <div id="toolbar">
+            <nimble-button></nimble-button>
+        </div>
+        <div>
+            <slot name="footer-actions"></slot>
+        </div>
+    </footer>
+</template>
+```
+
+An example usage of the `nimble-rich-text-editor` in the application layer is as follows:
+
+```html
+<nimble-rich-text-editor>
+    <div slot="footer-actions">
+        <nimble-button>Cancel</nimble-button>
+        <nimble-button>Add Comment</nimble-button>
+    </div>
+</nimble-rich-text-editor>
+```
+
+_nimble-rich-text-viewer_
+
+The `nimble-rich-text-viewer` is used for viewing rich text content when markdown string is passed to it. It performs the post-processing
+tasks to convert the markdown string to corresponding HTML nodes for each text formatting.
+
+### API
+
+#### `nimble-rich-text-editor`
+
+_Props/Attrs_
+
+-   none
+
+_Methods_
+
+-   `setMarkdown()` - sets the markdown input from the consumer component to preload the content in the editor.
+-   `getMarkdown()` - gets the processed markdown output from the component.
+-   [`destroy()`](https://tiptap.dev/api/editor#destroy) - stops the editor instance and unbinds all events.
```

Comment:
Any need for exposing `destroy`? Do we need to handle it internally on detaching this custom element?

Reply:
Yes, it can be handled internally inside the custom element by destroying the editor in `disconnectedCallback` lifecycle hooks.